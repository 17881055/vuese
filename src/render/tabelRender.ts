export function renderTabelHeader(header: string[]): string {
  const headerString = renderTabelRow(header)
  const splitLine = renderSplitLine(headerString.length)
  return headerString + splitLine + '\n'
}

export function renderTabelRow(row: string[]): string {
  return row.map(n => `|${n}`).join('') + '|\n'
}

function renderSplitLine(num: number): string {
  let line = ''
  for (let i = 0; i < num; i++) {
    line += '='
  }
  return line
}
