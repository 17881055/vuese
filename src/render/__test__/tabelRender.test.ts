import { renderTabelHeader } from '../tabelRender'

test('Proper rendering of the table header', () => {
  const header = ['Name', 'Description', 'Type', 'Required', 'Default']
  const str = renderTabelHeader(header)
  expect(str).toMatchSnapshot()
})
