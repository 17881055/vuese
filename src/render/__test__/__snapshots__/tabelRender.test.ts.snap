// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proper rendering of the table header 1`] = `
"|Name|Description|Type|Required|Default|
=========================================
"
`;
